pkg/kubecli/vm.go

```diff
@@ -20,23 +20,119 @@
 package kubecli
 
 import (
+	goerror "errors"
+	"fmt"
+	"io"
+
+	k8sv1 "k8s.io/api/core/v1"
 	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
 
 	"k8s.io/apimachinery/pkg/types"
 
 	"kubevirt.io/kubevirt/pkg/api/v1"
 )
 
 func (k *kubevirt) VM(namespace string) VMInterface {
-	return &vms{k.restClient, namespace, "virtualmachines"}
+	return &vms{
+		restClient: k.restClient,
+		clientSet:  k.Clientset,
+		namespace:  namespace,
+		resource:   "virtualmachines",
+		master:     k.master,
+		kubeconfig: k.kubeconfig,
+	}
 }
 
 type vms struct {
 	restClient *rest.RESTClient
+	clientSet  *kubernetes.Clientset
 	namespace  string
 	resource   string
+	master     string
+	kubeconfig string
+}
+
+func findPod(clientSet *kubernetes.Clientset, namespace string, name string) (string, error) {
+	fieldSelector := fields.ParseSelectorOrDie("status.phase==" + string(k8sv1.PodRunning))
+	labelSelector, err := labels.Parse(fmt.Sprintf(v1.AppLabel+"=virt-launcher,"+v1.DomainLabel+" in (%s)", name))
```

Comment:
Not for now, just in general.

I wonder if an ownership or object reference between vm and pod does make sense to make the lookup easier.

/cc @rmohr

Reply:
For garbage collection it very likely makes sense to add them (did not do it so far, since it needs to be a consistent story between the vm controller, the pod and the gc). Since it is not possible to look up objects based on owner references, labels are much easier.